### ISSUE TITLE
9.1.1版本 v-for="item in enquirySwipers" :key="item.inquiryId" 会报 "Expected 'v-bind:key' directive to use the variables which are defined by the 'v-for' directive" 的错误，在9.1.0版本就正常

### ISSUE THREAD

**Issue by huzhen-v5** [vuejs/vue-eslint-parser#188]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I'm using [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue).
- [X] I'm sure the problem is a parser problem. (If you are not sure, search for the issue in [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue) repo and open the issue in [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue) repo if there is no solution.
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

8.33.0

### What version of `eslint-plugin-vue` and ` vue-eslint-parser` are you using?

- vue-eslint-parser@9.1.1
- eslint-plugin-vue@9.9.0

### What did you do?

<details>
<summary>Configuration</summary>

```
<!-- Paste your configuration here -->
```
</details>

```vue
<swiper-slide
  v-for="item in enquirySwipers"
  :key="item.inquiryId"
>
  <div class="item-container">
    <img :src="item.inquiryImgUrl" alt="" />
  </div>
</swiper-slide>
```

### What did you expect to happen?

没有 vue/valid-v-for 的报错

### What actually happened?

eslint报错：Expected 'v-bind:key' directive to use the variables which are defined by the 'v-for' directive.eslint[vue/valid-v-for](https://eslint.vuejs.org/rules/valid-v-for.html)

### Link to Minimal Reproducible Example

null

**Comment by ota-meshi** [comment]
I don't know what kind of problem you have, but at least it doesn't seem to be a parser problem.

**Comment by chengzhuo5** [comment]
目前9.3.1版本都有这个问题，楼主怎么解决的？
